webapp/components/create_post.jsx

```diff
@@ -9,6 +9,7 @@ import FilePreview from './file_preview.jsx';
 import PostDeletedModal from './post_deleted_modal.jsx';
 import TutorialTip from './tutorial/tutorial_tip.jsx';
 import EmojiPicker from './emoji_picker/emoji_picker.jsx';
+import EditChannelHeaderModal from './edit_channel_header_modal.jsx';
```

Comment:
Instead of adding the modal here, I think you can use the one already added here https://github.com/mattermost/platform/blob/master/webapp/components/navbar.jsx

You'll want to update the modal store
https://github.com/mattermost/platform/blob/master/webapp/stores/modal_store.jsx#L33
to accept a new action type `TOGGLE_EDIT_CHANNEL_HEADER_MODAL`

Then in navbar.jsx just add a ModalStore event listener very similar to how it's done for the quick switch modal

Finally, you'll need an action in global_actions to dispatch the event to the modal store. Should be some examples in there already

Let me know if you need more clarification